core/network/src/test/mod.rs

```diff
@@ -662,6 +672,29 @@ pub trait TestNetFactory: Sized {
 			specialization,
 		).unwrap();
 
+		std::thread::spawn(move || {
```

Comment:
Does this thread just handle I/O and the protocol?

Reply:
Yes. The tests assume there's a background thread, and I went for the straight-forward fix, which is to put back a background thread.